src/osd/scheduler/mClockScheduler.cc

```diff
@@ -480,6 +538,8 @@ WorkItem mClockScheduler::dequeue()
       return std::move(*retn.request);
     }
   }
+  // We should not reach here, must have checked empty() first
+  return {};
```

Comment:
This can be removed as we are guaranteed to find an item from any of the queues.

Reply:
I think one of the unit tests had reached here because it didn't check empty() first. We never dequeue without checking empty() in other places. Should I modify the test?

Reply:
In the tests we always try to match the number of enqueue and dequeue operations. If the code reaches this point when you expect an item to be dequeued, then it probably indicates something amiss in the enqueue or dequeue logic.